apis/iap/v1alpha1/iapsettings_types.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var IAPSettingsGVK = GroupVersion.WithKind("IAPSettings")
+
+// IAPSettingsSpec defines the desired state of IAPSettings
+// +kcc:proto=google.cloud.iap.v1.IapSettings
+type IAPSettingsSpec struct {
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
+	// Immutable.
+	// The IAPSettings name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Required. The resource name of the IAP protected resource.
+	// +required
+	Name *string `json:"name,omitempty"`
+
+	// Top level wrapper for all access related setting in IAP
+	AccessSettings *AccessSettings `json:"accessSettings,omitempty"`
+
+	// Top level wrapper for all application related settings in IAP
+	ApplicationSettings *ApplicationSettings `json:"applicationSettings,omitempty"`
+}
+
+// IAPSettingsStatus defines the config connector machine state of IAPSettings
+type IAPSettingsStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the IAPSettings resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *IAPSettingsObservedState `json:"observedState,omitempty"`
+}
+
+// IAPSettingsSpec defines the desired state of IAPSettings
+// +kcc:proto=google.cloud.iap.v1.IapSettings
+type IAPSettingsObservedState struct {
```

Comment:
Suggest adding the configurable fields here as well (does IapSettings has any output-only field?)